spyder_kernels/customize/spyderpdb.py

```diff
@@ -214,11 +204,44 @@ def is_name_or_composed(text):
             text = text[1:]
             begidx += 1
 
-        matches = compfunc(text, line, begidx, endidx)
+        matches = []
+        if compfunc:
+            matches = compfunc(text, line, begidx, endidx)
+
+        cursor_start = cursor_pos - len(text)
+
+        if ipython_do_complete:
+            kernel = get_ipython().kernel
+            # Make complete call with current frame
+            if self.curframe:
+                kernel.shell.set_completer_frame(self.curframe)
+            result = super(SpyderKernel, kernel).do_complete(code, cursor_pos)
```

Comment:
I don't like this. Instead of importing `SpyderKernel` in this module and calling `super` here, please define a new method in `SpyderKernel` called `_do_complete`, like this

```python
def _do_complete(self, code, cursor_pos):
    """Call parent class do_complete"""
    return super(SpyderKernel, self).do_complete(code, cursor_pos)
```

With that we can replace this line with

```python
result = kernel._do_complete(code, cursor)
```

and you can also rewrite `do_complete` in `SpyderKernel` like this

```python
def do_complete(self, code, cursor_pos):
    """
    Call PdB complete if we are debugging.
    Public method of ipykernel overwritten for debugging.
    """
    if self.is_debugging():
        return self._pdb_obj.do_complete(code, cursor_pos)
    return self._do_complete(code, cursor_pos)
```